pytest_plugins/issue_handlers.py

```diff
@@ -0,0 +1,294 @@
+)
+
+COMPONENT = re.compile(
+    # To match :CaseComponent: FooBar
+    r"\s*:CaseComponent:\s*(?P<component>\S*)",
+    re.IGNORECASE,
+)
+
+IMPORTANCE = re.compile(
+    # To match :CaseImportance: Critical
+    r"\s*:CaseImportance:\s*(?P<importance>\S*)",
+    re.IGNORECASE,
+)
+
+BZ = re.compile(
+    # To match :BZ: 123456, 456789
+    r"\s*:BZ:\s*(?P<bz>.*\S*)",
+    re.IGNORECASE,
+)
+
+
+def _add_workaround(data, matches, usage, validation=(lambda *a, **k: True), **kwargs):
+    """Adds entry for workaround usage."""
+    for match in matches:
+        issue = f"{match[0]}:{match[1]}"
+        if validation(data, issue, usage, **kwargs):
+            data[issue.strip()]['used_in'].append({'usage': usage, **kwargs})
+
+
+def _should_deselect(issue, data=None):
+    """Check if test should be deselected based on marked issue."""
+    # Handlers can be extended to support different issue trackers.
+    handlers = {'BZ': bugzilla.should_deselect_bz}
+    supported_handlers = tuple(f"{handler}:" for handler in handlers.keys())
+    if str(issue).startswith(supported_handlers):
+        handler_code, _, _ = str(issue).partition(":")
+        return handlers[handler_code.strip()](issue.strip(), data)
+
+
+def generate_issue_collection(items, config):  # pragma: no cover
+    """Generates a dictionary with the usage of Issue blockers
+
+    For use in pytest_collection_modifyitems hook
+
+    Arguments:
+        items {list} - List of pytest test case objects.
+        config {dict} - Pytest config object.
+
+    Returns:
+        [List of dicts] - Dicts indexed by "<handler>:<issue>"
+
+        Example of return data::
+
+            {
+                "XX:1625783" {
+                    "data": {
+                        # data taken from REST api,
+                        "status": ...,
+                        "resolution": ...,
+                        ...
+                        # Calculated data
+                        "is_open": bool,
+                        "is_deselected": bool,
+                        "clones": [list],
+                        "dupe_data": {dict}
+                    },
+                    "used_in" [
+                        {
+                            "filepath": "tests/foreman/ui/test_sync.py",
+                            "lineno": 124,
+                            "testcase": "test_positive_sync_custom_ostree_repo",
+                            "component": "Repositories",
+                            "usage": "skip_if_open"
+                        },
+                        ...
+                    ]
+                },
+                ...
+            }
+    """
+    settings.configure()
+    valid_markers = ["skip_if_open", "skip", "deselect"]
+    collected_data = defaultdict(lambda: {"data": {}, "used_in": []})
+
+    bz_cache_filename = config.getvalue('bz_cache')  # use existing json cache?
+    cached_data = None
+    if bz_cache_filename is not None:
```

Comment:
```suggestion
    if bz_cache_filename:
```
as long as bz_cache_filename doesn't have any valid "falsey" values

Reply:
Its my preference to be explicit when checking against `None`, but I'm flexible if the shorter check is the syntax preference here in Robottelo.